src/libsyntax/parse/parser.rs

```diff
@@ -2571,7 +2569,7 @@ impl<'a> Parser<'a> {
                             s.print_usize(float.trunc() as usize)?;
                             s.pclose()?;
                             word(&mut s.s, ".")?;
-                            word(&mut s.s, fstr.splitn(2, ".").last().unwrap())
+                            word(&mut s.s, fstr.splitn(2, '.').last().unwrap())
```

Comment:
Wait a minute. I just benchmarked and found splitting by char be consistently faster on my machine, no matter the assembly.